### PR TITLE
collation: make gbk default collation to `gbk_bin` if new collation is not enabled

### PR DESCRIPTION
**Pull request by xiongjiwei**
### What problem does this PR solve?

Issue Number: close  https://github.com/pingcap/tidb/issues/31221.

make gbk default collation to gbk_utf8mb4_bin if new collation is not enabled

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

```release-note
None
```

### CODE REVIEW

File: parser/charset/encoding_gbk.go [hunk 1]
```diff
@@ -112,6 +112,14 @@ var GBKCase = unicode.SpecialCase{
 // customGBK is a simplifiedchinese.GBK wrapper.
 type customGBK struct{}
 
+func NewCustomGBKDecoder() *encoding.Decoder {
```
Comment (zimulala) [inline parser/charset/encoding_gbk.go:115]: We don't use it?

File: executor/executor_test.go [hunk 2]
```diff
@@ -5803,7 +5803,7 @@ func (s *testSuiteWithCliBaseCharset) TestCharsetFeature(c *C) {
 	tk.MustQuery("show charset").Check(testkit.Rows(
```
Comment (zimulala) [inline executor/executor_test.go:5803]: Please add this test to `cmd/explaintest/t/new_character_set.test`?

**Review by zimulala — APPROVED** [review]
LGTM

File: cmd/explaintest/t/new_character_set.test [hunk 3]
```diff
@@ -46,3 +46,5 @@ create table t (b blob, d json);
 insert into t values ('你好', '{"测试": "你好"}');
 select b, d from t;
 select hex(b), hex(d) from t;
+show charset;
+show collation;
```
Comment (tangenta) [inline cmd/explaintest/t/new_character_set.test:50]:
```suggestion
show collation;

```

**Review by Defined2014 — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Pull request merged**